Change many ERROR() calls into thrown exceptions

MrPerson opened the pull request:
This requires a 508 beta version to use. If Travis fails this he's a bitch

Exceptions will generate a stack trace, which is way easier to see and more helpful in actually solving this kind of crap. Also logs all the arguments, src, line, and file automatically.
Removed any dubiously helpful information in the exception names so the runtime condenser won't see each one as a different runtime. If the information is critical to solve these bugs (camera one maybe?), then I'll just make these warnings.
Thrown exceptions crash the currently running proc. Yes that means there's useless returns in a bunch of these, sue me. spawn()'s are to let the proc continue.

Almost all of these are difficult to trigger, but I did test playsound. And frankly even if they do cause bugs by crashing procs, big whoop

EDIT: Added a compile-time error if compiled in a pre-508 version of BYOND

MrPerson committed:
Change many ERROR() calls into thrown exceptions

This requires a 508 beta version to use. If Travis fails this he's a bitch

Exceptions will generate a stack trace, which is way easier to see and more helpful in actually solving this kind of crap. Also logs all the arguments, src, line, and file automatically.
Removed any dubiously helpful information in the exception names so the runtime condenser won't see each one as a different runtime. If the information is critical to solve these bugs (camera one maybe?), then I'll just make these warnings.
Thrown exceptions crash the currently running proc. Yes that means there's useless returns in a bunch of these, sue me. spawn()'s are to let the proc continue.

Almost all of these are difficult to trigger, but I did test playsound. And frankly even if they do cause bugs by crashing procs, big whoop

MrPerson committed:
Tell Travis to use a 508 beta version so it'll compile

MrPerson commented:
Duplicates #10788

Miauw62 commented on .travis.yml:
```diff
@@ -3,8 +3,8 @@ language: c
     global:
-    - BYOND_MAJOR="507"
-    - BYOND_MINOR="1286"
+    - BYOND_MAJOR="508"
```
It would be nice if there was a compiletime error or warning outside of travis when compiling with pre-507 versions, since that throws a bad text error in malfunction.dm and people keep making issues about how the code doesn't compile while it does.

MrPerson replied:
Done

Miauw62 commented on code/datums/ai_laws.dm:
```diff
@@ -92,16 +93,16 @@
 				add_inherent_law("You may not injure a human being or, through inaction, allow a human being to come to harm.")
 				add_inherent_law("You must obey orders given to you by human beings, except where such orders would conflict with the First Law.")
 				add_inherent_law("You must protect your own existence as long as such does not conflict with the First or Second Law.")
+				throw EXCEPTION("Invalid custom AI laws, check silicon_laws.txt")
```
This should probably be a WARNING instead of an exception.

Miauw62 commented on code/game/gamemodes/gang/gang_datum.dm:
```diff
@@ -21,7 +21,8 @@
 	if(!gang_colors_pool.len)
 		message_admins("WARNING: Maximum number of gangs have been exceeded!")
-		ERROR("WARNING: Maximum number of gangs have been exceeded!")
+		throw EXCEPTION("Maximum number of gangs have been exceeded")
```
Should be has, not have :v

MrPerson commented:
It occurs to me I could (should) have kept warning() and just had it throw an exception, which wouldn't crash the calling proc

WELP TOO LATE NOW HA HA

MrPerson committed:
Add a compile time warning for out of date byond versions
Of course none of you idiots are going to upkeep this shit so I'm not sure why I even bother

Also fixes the stuff pointed out by Miauw
